core/src/components/datetime/datetime.tsx

```diff
@@ -1409,6 +1434,18 @@ export class Datetime implements ComponentInterface {
     this.emitStyle();
   }
 
+  private errorIfTimeZoneProvided() {
+    const { dateFormatOptions, timeFormatOptions } = this;
+    if (
+      dateFormatOptions?.timeZone ||
+      dateFormatOptions?.timeZoneName ||
+      timeFormatOptions?.timeZone ||
+      timeFormatOptions?.timeZoneName
+    ) {
+      printIonWarning('Datetime: "timeZone" and "timeZoneName" are not supported in "formatOptions".');
```

Comment:
We'll also want to strip out `timeZoneName` from the options. `timeZone` should be handled by manually adding our own in the util functions, but `timeZoneName` is kept.

It may make the most sense to strip out both here, to keep things easy to understand and ensure we always avoid using `timeZone` even if we remove the manual UTC addition later.

Reply:
There are also a few more warnings we should log if possible; from the design doc:
> For presentations `date`, `month`, `month-year`, and `year`, only the `date` piece is required. If it is not included, the default `date` options should be used and a warning should be logged. If the `time` piece is included, it will be ignored.
>
> For the `time` presentation, only the `time` piece is required. If it is not included, the default `time` options should be used and a warning should be logged. If the `date` piece is included, it will be ignored.
>
> For presentations `date-time` and `time-date`, both pieces are required. If only a `date` or `time` object is provided (e.g. `{ date: { ... } }`), the default value should be used for the missing piece, and a warning should be logged to the developer.

It might be easiest to check for these conditions and log appropriate warnings here. This could alternatively be done in the util functions where the default is actually fallen back to.

Reply:
Oh, I have the other warnings in my datetime-button branch, I think because the presentations were called out under datetime-button. But I can move them here, if you prefer?

Reply:
Ah okay, that's fine then 👍 Just the time zone stuff in that case.

Reply:
fixed the time zone stuff in c021dc5995

Reply:
i went ahead and added the other logs to this PR in 3f61fe16b4